Use normal.cross(Vector(0,0,-1)) for Plane(face) x_dir to ensure consistent orientation

javimixet opened the pull request:
### Summary
Use a consistent geometric rule for determining `x_dir` in `Plane(face)` instead of relying on the face surface’s U-derivative.

### Details
When `Plane` is created from a `Face` without an explicit `x_dir`, the X direction is now computed as:

```python
x_dir = face.normal_at().cross(Vector(0, 0, -1))
```

- For horizontal faces, X aligns with global +X.  
- For vertical/sloped faces, X lies parallel to the global XY plane and points **right** to an observer standing in front of the face.  

This produces predictable, human-intuitive orientations and avoids inconsistent X directions between adjacent faces.

javimixet committed:
Plane(face) x_dir handling

Change to use `normal.cross(Vector(0,0,-1))` instead of the u derivative of the surface for x_dir when x_dir is not given.

javimixet committed:
Update direction assertions in test_plane.py

Update tests for new Plane(face) x_dir normal.cross(-Z) behavior.

javimixet committed:
Change orientation assertion in test_plane.py

Updated orientation assertion to check against fixed values.

javimixet committed:
Update orientation assertion to use Vector

javimixet committed:
Fix coordinates in Locations context

javimixet committed:
Merge branch 'gumyr:dev' into Plane_face_x_dir_default

gumyr commented on src/build123d/geometry.py:
```diff
@@ -2810,25 +2810,21 @@ def __init__(self, *args, **kwargs):
+                face_x_dir = Vector(1, 0, 0)
             else:
-                face_x_dir = surface.Position().XDirection()
+                face_x_dir = face_z_dir.cross(Vector(0, 0, -1)).normalized()
```
I don't think this works.  If the `face_z_dir` is (0,0,-1) the cross product is (0,0,0) which will cause this to fail.

javimixet replied:
Don't lines 2819 and 2820 handle that?

gumyr replied:
Not mathematically but practically maybe yes.  I'm concerned that there might be magic number stuff happening when there are small differences in the z direction calculation.

gumyr reviewed (commented):
I'm not a fan of trying to fix this "problem". If there is not enough information to create the plane then the user shouldn't expect a known outcome.  One can provide an x_dir when creating a Plane from a Face or an Axis which allows the user total control over the resulting Plane.

javimixet commented:
> I'm not a fan of trying to fix this "problem". If there is not enough information to create the plane then the user shouldn't expect a known outcome. One can provide an x_dir when creating a Plane from a Face or an Axis which allows the user total control over the resulting Plane.

That makes sense, and I agree this isn’t a bug in the current behavior. My intent wasn’t to “fix” anything, but to suggest a slightly more predictable default when x_dir isn’t provided.

Users already have full control by explicitly passing an x_dir, and this wouldn’t change that. I see it as a small default enhancement, in line with how build123d already handles many defaults—but of course, it’s your call, and I’m completely fine if you’d rather keep things as they are.

javimixet committed:
Refactor face direction calculation in geometry.py

Switched to checking the cross-product magnitude directly instead of relying on a Z-component threshold, so we avoid normalizing near-zero vectors and reduce tolerance-sensitive behavior.
